backend/mqttauth/views.py

```diff
@@ -1,62 +1,70 @@
 import logging
 
 from django.conf import settings
-from django.contrib.auth import authenticate, get_user_model
+from django.contrib.auth import get_user_model
 from django.http import HttpResponse
 from django.views.decorators.csrf import csrf_exempt
+from rest_framework import status
 
+from dhh_auth.models import ClientPermission
+from dhh_auth.utils import get_perm_obj
 from gatekeeper.models import Datastream
 from gatekeeper.utils import parse_sta_url
+from service.models import Service, ServiceToken
 
 logger = logging.getLogger(__name__)
 
 
-# TODO: Should be moved to a custom user manager when we have one
-def get_user_by_username_or_anonymous(username):
-    user_class = get_user_model()
-    try:
-        return user_class.objects.get(username=username, is_active=True)
-    except user_class.DoesNotExist:
-        return user_class.get_anonymous()
-
-
 @csrf_exempt
 def auth(request):
     """Check username/password
 
     URI-Param           username   password    clientid    topic   acc
     http_getuser_uri    Y          Y           N           N       N
     """
-    response_status_code = 403
+    response_status_code = status.HTTP_403_FORBIDDEN
 
-    user = authenticate(request, username=request.POST.get('username'), password=request.POST.get('password'))
+    try:
+        ServiceToken.objects.get(service__identifier=request.POST.get('username'), service__is_active=True,
+                                 key=request.POST.get('password'))
 
-    # For now all registered users can connect
-    if user:
-        response_status_code = 200
+        response_status_code = status.HTTP_200_OK
+    except ServiceToken.DoesNotExist:
+        pass
 
-    logger.info('MQTT authentication for user "{}" {}'.format(
-        request.POST.get('username'), 'succeeded' if response_status_code == 200 else 'failed'))
+    logger.info('MQTT authentication for service "{}" {}'.format(
+        request.POST.get('username'), 'succeeded' if response_status_code == status.HTTP_200_OK else 'failed'))
 
     return HttpResponse(status=response_status_code)
 
 
 @csrf_exempt
 def superuser(request):
-    """Check superuser
+    """Check if the user with the supplied username is a superuser
+
+    Excludes any username that is an identifier of a service because
+    they could be mixed up and could end in a privilege escalation.
 
     URI-Param           username   password    clientid    topic   acc
     http_superuser_uri  Y          N           N           N       N
     """
-    response_status_code = 403
+    response_status_code = status.HTTP_403_FORBIDDEN
+
+    username = request.POST.get('username')
+    user = None
 
-    user = get_user_by_username_or_anonymous(request.POST.get('username'))
+    user_class = get_user_model()
+    try:
+        service_identifiers = Service.objects.all().values_list('identifier', flat=True)
+        user = user_class.objects.exclude(username__in=service_identifiers).get(username=username, is_active=True)
```

Comment:
Hmm.. this is a bit of a shortcoming with the current `Service` setup it seems. Might be good to have a `is_superuser` in the `Service` model as well so that lookups like this does not have to be made. This would mean that a `User` instance has to exist with the same name as a `Service` which is not nice. What do you think?

Reply:
The super user permissions are only needed for the mosquitto broker client. I don't think we want to give such permissions to any services.

Reply:
Yeah, that is probably true indeed! No need for any changes 😄